FinSearchUnified/Tests/Bundle/SearchBundleFindologic/FacetHandler/TextFacetHandlerTest.php

```diff
@@ -0,0 +1,360 @@
+use Shopware\Bundle\SearchBundle\ConditionInterface;
+use Shopware\Bundle\SearchBundle\Criteria;
+use Shopware\Bundle\SearchBundle\Facet\ProductAttributeFacet;
+use Shopware\Bundle\SearchBundle\FacetResult\RadioFacetResult;
+use Shopware\Bundle\SearchBundle\FacetResult\ValueListFacetResult;
+use Shopware\Bundle\SearchBundle\FacetResult\ValueListItem;
+use Shopware\Components\Test\Plugin\TestCase;
+use SimpleXMLElement;
+
+class TextFacetHandlerTest extends TestCase
+{
+    /**
+     * @dataProvider filterProvider
+     *
+     * @param string $name
+     * @param string $type
+     * @param bool $doesSupport
+     */
+    public function testSupportsFilter($name, $type, $doesSupport)
+    {
+        $data = '<?xml version="1.0" encoding="UTF-8"?><searchResult></searchResult>';
+        $filter = new SimpleXMLElement($data);
+
+        $filter->addChild('name', $name);
+        $filter->addChild('type', $type);
+
+        $facetHandler = new TextFacetHandler();
+        $result = $facetHandler->supportsFilter($filter);
+
+        $this->assertSame($doesSupport, $result);
+    }
+
+    public function filterProvider()
+    {
+        return [
+            'Category filter with "select" type' => ['cat', 'select', false],
+            'Category filter with "label" type' => ['cat', 'label', false],
+            'Vendor filter with "select" type' => ['vendor', 'label', true],
+            'Vendor filter with "label" type' => ['vendor', 'label', true],
+        ];
+    }
+
+    public function testFacetModeForBooleanResult()
+    {
+        $data = '<?xml version="1.0" encoding="UTF-8"?><searchResult></searchResult>';
+        $filter = new SimpleXMLElement($data);
+
+        $facet = new ProductAttributeFacet(
+            'field',
+            ProductAttributeFacet::MODE_BOOLEAN_RESULT,
+            'field',
+            'Field'
+        );
+
+        $facetHandler = new TextFacetHandler();
+        $result = $facetHandler->generatePartialFacet($facet, new Criteria(), $filter);
+
+        $this->assertNull($result);
+    }
+
+    /**
+     * @dataProvider valueListDataProvider
+     *
+     * @param array $filterData
+     * @param ValueListFacetResult $facetResult
+     * @param ConditionInterface|null $condition
+     */
+    public function testFacetModeForListFacetResult(
+        array $filterData,
+        ValueListFacetResult $facetResult,
+        ConditionInterface $condition = null
+    ) {
+        $facet = new ProductAttributeFacet(
+            'vendor',
+            ProductAttributeFacet::MODE_VALUE_LIST_RESULT,
+            'vendor',
+            'Manufacturer'
+        );
+        $criteria = new Criteria();
+        if ($condition !== null) {
+            $criteria->addCondition($condition);
+        }
+
+        $data = '<?xml version="1.0" encoding="UTF-8"?><searchResult></searchResult>';
+        $filter = new SimpleXMLElement($data);
+        foreach ($filterData as $key => $value) {
```

Comment:
Tests should be simple, this is way to complex. We need a test to test the test if it is to complex.

Reply:
At least move this in its own function called like 'generateFilter()', also you could reuse this new function in line 137.

Reply:
@wladi0097 sure, we can do that. 
thanks